docs/rfcs/rfc-040-relay-first-api.md

```diff
@@ -0,0 +1,1351 @@
+                title
+            }
+        }
+    }
+}
+```
+
+> Note that the plain `fulltext` option without `where` may still filter the results.
+
+## Simple API - Query
+
+Relay API, while complete and flexible, is verbose. This may affect onboarding and simple use cases. For this cases, a simplified API can be used. This API is more limited and **cannot** be used along with the Relay API.
+
+The top level queries will be named after the types provided:
+
+```graphql
+type Query {
+    movies(where: MovieWhere, sort: [MovieSortNode]): [Movie!]!
+    people(where: PersonWhere, sort: [PersonSortNode]): [Person!]!
+}
+```
+
+Instead of a custom `*Connection` type, these queries return the type provided by the user directly. This API can be queried as follows:
+
+```graphql
+query Movies {
+    movies {
+        title
+    }
+}
+```
+
+Nested queries are also supported:
+
+```graphql
+query Movies {
+    movies {
+        title
+        actors {
+            name
+        }
+    }
+}
+```
+
+### Filtering
+
+Filtering is supported with all of the operators available in the Connections API. Filtering over edges, aggregations or fulltext is not supported.
+
+```graphql
+query Movies {
+    movies(
+        where: { OR: [{ title: { equals: "The Matrix" } }, { actors: { some: { name: { contains: "Keanu" } } } }] }
+    ) {
+        title
+        actors {
+            name
+        }
+    }
+}
+```
+
+### Sort and Pagination
+
+Sort behaves in the same fashion as the Connections API. Cursor-based pagination, however, is not supported. The simple API uses _offset_ based pagination:
+
+```graphql
+query Movies {
+    movies(sort: { title: DESC }, limit: 10, offset: 2) {
+        title
+    }
+}
+```
+
+### Limitations
+
+-   Edge fields cannot be queried or filtered
+-   Aggregations are not available
+-   Cursor based pagination is not supported
+-   Fulltext is not supported
+
+## Naming conventions
+
+These are some loose conventions on type and input naming:
+
+-   `Connection`: Types containing `edges` and following Relay spec.
+-   `Edge`: Types containing `node` and, optionally, fields
+-   `Where`: Filtering Types
+-   `Aggregation`: Aggregation Types
+
+*   Types in a relationship will follow the convention `[RootType][fieldName]` at the beginning.
+*   Where possible, input types will mirror normal types. E.g. `PersonNode` will be filtered by `PersonNodeWhere`
+*   Types ending in `*Connection` will follow Relay spec as these are reserved types according to the spec
+
+### Further discussion points
+
+-   `Fields` in edges vs having the fields directly on top-level
+-   Uppercase operators (e.g. `AND: [{title: CONTAINS: "Matrix"}]`)
+-   Remove top level edge from `where` and mutations
+-   `fulltextScore` vs `fulltext { score }`
+-   Connections may have a `nodes` fields as a shortcut without passing through `edges`
+-   Sort by aggregations and 1-\* relationship
+-   Union / interfaces
+-   Nested [field]connection
+
```

Comment:
I just noticed, that one thing that we don't analyze at all during the process (for good reason, as the scope is already large like it is now) is the Subscription related types.
Could it be worth adding it as a discussion point?